src/plugins/idents.ts

```diff
@@ -85,41 +85,42 @@ function defaultShouldAlias(name: string, freq: number): boolean {
 }
 export function aliasBuiltins(
   shouldAlias: (name: string, freq: number) => boolean = defaultShouldAlias
-): Visitor {
-  const usedBuiltins = new Map<string, Identifier[]>();
+): Plugin {
   return {
     name: "aliasBuiltins(...)",
-    enter(path: Path) {
-      const node = path.node;
-      if (node.kind === "Identifier" && node.builtin) {
-        if (!usedBuiltins.has(node.name)) usedBuiltins.set(node.name, []);
-        usedBuiltins.get(node.name)?.push(node);
-      }
-    },
-    exit(path: Path) {
-      if (path.parent === null) {
-        const program = path.root.node;
-        const aliased: string[] = [];
-        usedBuiltins.forEach((identifiers: Identifier[], name: string) => {
-          if (shouldAlias(name, identifiers.length)) {
-            aliased.push(name);
-            for (const ident of identifiers) {
-              ident.builtin = false;
-              ident.name = name + "_alias";
-            }
-          }
-        });
-        program.body = block(
-          aliased
-            .map((x) => assignment(x + "_alias", id(x, true)) as Expr)
-            .concat(
-              program.body.kind === "Block"
-                ? program.body.children
-                : [program.body]
-            )
-        );
-        usedBuiltins.clear();
+    visit(program, spine) {
+      if (program.kind !== "Program") return;
+      // get frequency of each builtin
+      const timesUsed = new Map<string, number>();
+      for (const name of spine.compactMap((node) => {
+        if (node.kind === "Identifier" && node.builtin) return node.name;
+      })) {
+        const freq = timesUsed.get(name) ?? 0;
+        timesUsed.set(name, freq + 1);
       }
+      // apply
+      const assignments: (IR.Assignment & { variable: Identifier })[] = [];
+      const replacedDeep = spine.withReplacer((node) => {
+        if (
+          node.kind === "Identifier" &&
+          node.builtin &&
```

Comment:
Should this be `!node.builtin`?

Reply:
This is `aliasBuiltins`, so the `node.builtin` check is to ensure it's only aliasing builtins. I guess that's because non-builtins (essentially) always have a one-letter name.